### ISSUE TITLE
unsubscriptable-object error reported for abstract generic types

### ISSUE THREAD

**Issue by aviv-ebates** [PyCQA/pylint#2822]
(This might be a part of #2416, but I already have those changes)
Pylint reports `unsubscriptable-object` for _abstract_ generic types:

### Steps to reproduce
Run pylint on following file:
```py3
from typing import TypeVar, Generic
from abc import ABCMeta, abstractmethod

T = TypeVar('T')

class Factory(Generic[T], metaclass=ABCMeta):
    @abstractmethod
    def build(self) -> T:
        pass

class IntFactory(Factory[int]):  # Sadpanda unsubscriptable-object
    def build(self) -> int:
        return 3

factory = IntFactory()

x: int = factory.build()
```

### Current behavior
Pylint reports `E1136: Value 'Factory' is unsubscriptable (unsubscriptable-object)` on line 11.

### Expected behavior
As far as I understand, this is supposed to be ok? `mypy 0.660` doesn't complain and python 3.7.1 runs the whole thing.

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.7.1 (default, Oct 22 2018, 11:21:55)
[GCC 8.2.0]
```

**Comment by PCManticore** [comment]
Thanks @aviv-ebates I can confirm the bug.

**Comment by santanaraphael** [comment]
I am having a similar issue, was this problem fixed for anyone?

**Comment by PCManticore** [comment]
@santanaraphael This is still an issue, no one got to work on it just yet.

**Comment by lovasoa** [comment]
Any update on this ?

**Comment by halvorboe** [comment]
Having the same problem.

**Comment by dariobig** [comment]
+1

**Comment by gibiansky** [comment]
I think this doesn't have to do with abstract generic types. I can reproduce this error with just a metaclass:

```python
import typing                                     
                                                  
variable = typing.TypeVar("variable")                                                   
                                                  
class A(typing.Generic[variable], metaclass=type):
    pass                                        
                                                  
class B(A[variable]):                             
    pass                                          
```

This might have to do with the fact that `typing.Generic[variable].__class__` is `typing._GenericAlias`, not `type`, but I couldn't figure out where to go from there.

**Comment by zuckerruebe** [comment]
+1. Cheers.

**Comment by ShadowLNC** [comment]
Not sure if this is related, but it also does not seem to accept the builtin value `type` as being subscriptable.

**Comment by cdce8p** [comment]
This issue will be fixed with the next release of pylint.
